libraries/Microsoft.Bot.Streaming/Utilities/Background.cs

```diff
@@ -70,7 +72,9 @@ public static void RunForever(Func<CancellationToken, Task<TimeSpan>> task, Time
                     {
                         spanDelay = await task(token).ConfigureAwait(false);
                     }
+#pragma warning disable CA1031 // Do not catch general exception types
                     catch (Exception)
+#pragma warning restore CA1031 // Do not catch general exception types
                     {
                     }
```

Comment:
why swallow?

Reply:
If we could pipe an  Ilogger in and use properties/eventname the way they were intended, to log failures, we could clean this entire class up a bit. With the IDisposable chain now in place we could actually handle exceptions thrown here and clean up the assemblers/disassemblers that would be derailed by them, too. In either case though we're courting behavior changes that would need more testing than the improvements on this PR, so I'd ask we not block on it for now.